### PR TITLE
DOC: sparse.linalg.gcrotmk: fix backticks and add `maxiter` default

### PR DESCRIPTION
**Pull request by anushkasuyal**
Added the description of the default value for the 'maxiter' parameter in the gcrotmk documentation to provide complete information.

### CODE REVIEW

**Comment by lucascolley** [comment]
okay, looks like both the backtick and the maxiter default changes are in this PR, that's fine 👍

File: scipy/sparse/linalg/_isolve/_gcrotmk.py [hunk 1]
```diff
@@ -221,10 +221,10 @@ def gcrotmk(A, b, x0=None, *, rtol=1e-5, atol=0., maxiter=1000, M=None, callback
         According to [2]_, good values are around m.
         Default: m
     CU : list of tuples, optional
-        List of tuples ``(c, u)`` which contain the columns of the matrices
+        List of tuples `(c, u)` which contain the columns of the matrices
         C and U in the GCROT(m,k) algorithm. For details, see [2]_.
         The list given and vectors contained in it are modified in-place.
-        If not given, start from empty matrices. The ``c`` elements in the
+        If not given, start from empty matrices. The `c` elements in the
```
Comment (j-bowhay) [inline scipy/sparse/linalg/_isolve/_gcrotmk.py:227]: This is not a argument to the function so was correct before
```suggestion
        If not given, start from empty matrices. The ``c`` elements in the
```

File: scipy/sparse/linalg/_isolve/_gcrotmk.py [hunk 2]
```diff
@@ -221,10 +221,10 @@ def gcrotmk(A, b, x0=None, *, rtol=1e-5, atol=0., maxiter=1000, M=None, callback
         According to [2]_, good values are around m.
         Default: m
     CU : list of tuples, optional
-        List of tuples ``(c, u)`` which contain the columns of the matrices
+        List of tuples `(c, u)` which contain the columns of the matrices
```
Comment (j-bowhay) [inline scipy/sparse/linalg/_isolve/_gcrotmk.py:224]: same here
```suggestion
        List of tuples ``(c, u)`` which contain the columns of the matrices
```

File: scipy/sparse/linalg/_isolve/_gcrotmk.py [hunk 3]
```diff
@@ -200,19 +200,19 @@ def gcrotmk(A, b, x0=None, *, rtol=1e-5, atol=0., maxiter=1000, M=None, callback
...
-        inverse of A. gcrotmk is a 'flexible' algorithm and the preconditioner
+        Preconditioner for `A`.  The preconditioner should approximate the
+        inverse of `A`. gcrotmk is a 'flexible' algorithm and the preconditioner
         can vary from iteration to iteration. Effective preconditioning
         dramatically improves the rate of convergence, which implies that
         fewer iterations are needed to reach a given error tolerance.
     callback : function, optional
         User-supplied function to call after each iteration.  It is called
-        as callback(xk), where xk is the current solution vector.
+        as `callback(xk)`, where `xk` is the current solution vector.
```
Comment (j-bowhay) [inline scipy/sparse/linalg/_isolve/_gcrotmk.py:215]: same here
```suggestion
        as ``callback(xk)``, where ``xk`` is the current solution vector.
```

File: scipy/sparse/linalg/_isolve/_gcrotmk.py [hunk 4]
```diff
@@ -200,19 +200,19 @@ def gcrotmk(A, b, x0=None, *, rtol=1e-5, atol=0., maxiter=1000, M=None, callback
     rtol, atol : float, optional
         Parameters for the convergence test. For convergence,
         ``norm(b - A @ x) <= max(rtol*norm(b), atol)`` should be satisfied.
-        The default is ``rtol=1e-5``, the default for ``atol`` is ``0.0``.
+        The default is ``rtol=1e-5``, the default for `atol` is ``0.0``.
```
Comment (j-bowhay) [inline scipy/sparse/linalg/_isolve/_gcrotmk.py:203]: For consistency
```suggestion
        The default is ``rtol=1e-5`` and ``atol=0.0``.
```

**Review by j-bowhay — CHANGES_REQUESTED** [review]
Thanks @anushkasuyal some suggestions

File: scipy/sparse/linalg/_isolve/_gcrotmk.py (hunk 2 above)
Comment (lucascolley) [inline scipy/sparse/linalg/_isolve/_gcrotmk.py:224]:
```suggestion
        List of tuples ``(c, u)`` which contain the columns of the matrices
```

`(c, u)` isn't a parameter name (rather a representation of an argument), so we want double backticks here too

File: scipy/sparse/linalg/_isolve/_gcrotmk.py (hunk 1 above)
Comment (lucascolley) [inline scipy/sparse/linalg/_isolve/_gcrotmk.py:227]:
```suggestion
        If not given, start from empty matrices. The ``c`` elements in the
```

File: scipy/sparse/linalg/_isolve/_gcrotmk.py [hunk 5]
```diff
@@ -221,10 +221,10 @@ def gcrotmk(A, b, x0=None, *, rtol=1e-5, atol=0., maxiter=1000, M=None, callback
         According to [2]_, good values are around m.
         Default: m
```
Comment (lucascolley) [inline scipy/sparse/linalg/_isolve/_gcrotmk.py:222]:
```suggestion
        Default: `m`
```

**Review by lucascolley — CHANGES_REQUESTED** [review]
this looks good to me after the suggested changes are applied!

**Comment by lucascolley** [comment]
oops, sorry for the double review! I'll hide my comments and check that I'm happy with Jake's

File: scipy/sparse/linalg/_isolve/_gcrotmk.py (hunk 5 above)
Comment (lucascolley) [reply scipy/sparse/linalg/_isolve/_gcrotmk.py]: could you add single backticks around `m` here?

```
        According to [2]_, good values are around `m`.
        Default: `m`
```

Comment (anushkasuyal) [reply scipy/sparse/linalg/_isolve/_gcrotmk.py]: done!

**Review by lucascolley — APPROVED** [review]
all good now, thanks and congrats on your first contribution to SciPy!

**Pull request merged**